libs/core/tests/unit_tests/test_tools.py

```diff
@@ -628,7 +628,7 @@ def test_named_tool_decorator_return_direct() -> None:
     """Test functionality when arguments and return direct are provided as input."""
 
     @tool("search", return_direct=True)
-    def search_api(query: str, *args: Any) -> str:
+    def search_api(query: str) -> str:
```

Comment:
No modifications to unrelated code should be made unless original code was wrong

```suggestion
    def search_api(query: str, *args: Any) -> str:
```

Reply:
In fact, this change is necessary for the test to pass after the proposed change.
As I mentioned in the PR body, I see no clean way to have my new test and this old test pass without changes to the old test.

I opted for changing the existing test, because adding `*args` to an `@tool` decorated method is a strange thing to do. It means the LLM is instructed to add arbitrary additional `args` without any instructions on how the LLM should write these. I think it is inadvisable to write a tool in such a manner, and hence I think its acceptable for this PR to break that kind of usage of the @tool decorator.

Of course, having full backwards compatibility would be much preferred, so let me know if you see a way to get that.